ibis/mapd/client.py

```diff
@@ -0,0 +1,326 @@
+
+    def _execute(self, query, results=False):
+        """
+
+        :param query:
+        :return:
+        """
+        if not self.execution_type == 3:
+            raise NotImplemented()
+
+        return MapDCursor(self.con.cursor().execute(query))
+
+    def database(self, name=None):
+        """Connect to a database called `name`.
+
+        Parameters
+        ----------
+        name : str, optional
+            The name of the database to connect to. If ``None``, return
+            the database named ``self.current_database``.
+
+        Returns
+        -------
+        db : Database
+            An :class:`ibis.client.Database` instance.
+
+        Notes
+        -----
+        This creates a new connection if `name` is both not ``None`` and not
+        equal to the current database.
+        """
+        if name == self.current_database or (
+            name is None and name != self.current_database
+        ):
+            return self.database_class(self.current_database, self)
+        else:
+            client_class = type(self)
+            new_client = client_class(
+                uri=self.uri, user=self.user, password=self.password,
+                host=self.host, port=self.port, dbname=name,
+                protocol=self.protocol, execution_type=self.execution_type
+            )
+            return self.database_class(name, new_client)
+
+    @property
+    def current_database(self):
+        return self.dbname
+
+    def set_database(self, name):
+        raise NotImplementedError(
+            'Cannot set database with MapD client. To use a different'
+            ' database, use client.database({!r})'.format(name)
+        )
+
+    def exists_database(self, name):
+        raise NotImplementedError()
+
+    def list_databases(self, like=None):
+        raise NotImplementedError()
+
+    def exists_table(self, name: str, database: str=None):
+        """
+        Determine if the indicated table or view exists
+
+        Parameters
+        ----------
+        name : string
+        database : string, default None
+
+        Returns
+        -------
+        if_exists : boolean
+        """
+        return len(self.list_tables(like=name, database=database)) > 0
+
+    def list_tables(self, like=None, database=None):
+        return self.con.get_tables()
+
+    def get_schema(self, table_name, database=None):
+        """
+        Return a Schema object for the indicated table and database
+
+        Parameters
+        ----------
+        table_name : string
+          May be fully qualified
+        database : string, default None
+
+        Returns
+        -------
+        schema : ibis Schema
+        """
+        col_names = []
+        col_types = []
+
+        for col in self.con.get_table_details(table_name):
+            col_names.append(col.name)
+            col_types.append(MapDDataType.parse(col.type))
+
+        return sch.schema(col_names, col_types)
```

Comment:
This can be more concise if you write it as `sch.schema([(col.name, MapDDataType.parse(col.type)) for col in self.con.get_table_details(table_name)])`

Reply:
Of course, insert some line breaks as necessary to make sure you pass `flake8`